traffic_portal/app/src/modules/private/profiles/compare/diff/index.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+module.exports = angular.module('trafficPortal.private.profiles.diff', [])
+	.config(function($stateProvider, $urlRouterProvider) {
+		$stateProvider
+			.state('trafficPortal.private.profiles.diff', {
+				url: '/{profile1Id:[0-9]{1,8}}/{profile2Id:[0-9]{1,8}}/compare/diff',
+				views: {
+					profilesContent: {
+						templateUrl: 'common/modules/table/profilesParamsCompare/table.profilesParamsCompare.tpl.html',
+						controller: 'TableProfilesParamsCompareController',
+						resolve: {
+							profile1: function($stateParams, profileService) {
+								return profileService.getProfile($stateParams.profile1Id, { includeParams: true });
+							},
+							profile2: function($stateParams, profileService) {
+								return profileService.getProfile($stateParams.profile2Id, { includeParams: true });
+							},
+							profilesParams: function(profile1, profile2) {
+								// funky way to do a unique (by id) union of param objects
+								return _.chain(profile1.params).union(profile2.params).unique('id').value();
```

Comment:
I don't know what this could possibly be doing. An underscore-wrapped array of hashes has no property `unique`, and I can't find anything like what you're doing in [the underscorejs documentation](https://underscorejs.org/) so I don't think it's wise to rely on this. Here's what I would do instead:
```javascript
const ids = new Set();
const uniq = new Array();
for (const param of profile1.params.concat(profile2.params)) {
    if (!ids.has(param.id)) {
        uniq.push(param);
    }
}
return uniq;
```

Reply:
Alternatively:
```
const ids = new Set();
return profile1.params.concat(profile2.params).filter((x)=>{if (!ids.has(x.id)){ids.add(x.id);return true;} return false;});
```
but I think that's a little too messy for a one-liner.